src/index.cpp

```diff
@@ -837,6 +837,31 @@ template <typename T, typename TagT, typename LabelT> std::vector<uint32_t> Inde
     return init_ids;
 }
 
+template <typename T, typename TagT, typename LabelT>
+const std::vector<LabelT> Index<T, TagT, LabelT>::find_common_filters(uint32_t point_id, bool search_invocation,
+                                                                      const std::vector<LabelT> &incoming_labels)
+{
+    std::vector<LabelT> common_filters;
+    auto &curr_node_labels = _pts_to_labels[point_id];
+    std::set_intersection(incoming_labels.begin(), incoming_labels.end(), curr_node_labels.begin(),
```

Comment:
Depending on the number of labels we expect to see, this might become very slow. The question is whether to change it upfront assuming that we might deal with large # of labels at some point in time, or deal with it at that time. Just a comment.

Reply:
Siddharth and Ravi are already working on this to replace with bit wise and operation.
Closing this comment in the context of this PR.